cni/pkg/cmd/root.go

```diff
@@ -100,11 +102,23 @@ var rootCmd = &cobra.Command{
 			// as well as listen for messages from the CNI binary.
 			cniEventAddr := filepath.Join(cfg.InstallConfig.CNIAgentRunDir, constants.CNIEventSocketName)
 			log.Infof("Starting ambient node agent with inpod redirect mode on socket %s", cniEventAddr)
+
+			// instantiate and validate the ambient enablement selector
+			lsCfg := &metav1.LabelSelector{}
```

Comment:
I think we will want a `[]etav1.LabelSelector` not a single one. This allows you to do OR operations.

I don't think you can actually represent the current semantics without a list either, which is probably handy to represent the default in terms of a `LabelSelector` instead of a special case?

Reply:
Ah sorry, actually i mean

```
enabled:
- namespaces: # type metav1.LabelSelector
  pods: # type metav1.LabelSelector
```

You can set pods, namespaces, or both. Then we can really define our existing logic as one. This meets all the use cases such as "Do not allow namespaces to opt out" etc